Sources/BraintreePayPal/BTPayPalClient.swift

```diff
@@ -273,37 +277,45 @@ import BraintreeDataCollector
-                            self.apiClient.sendAnalyticsEvent(eventName)
-                        } else {
-                            // User tapped system cancel button on permission alert
-                            let eventName = "ios.\(paymentType.stringValue).authsession.alert.cancel"
-                            self.apiClient.sendAnalyticsEvent(eventName)
-                        }
-                    }
-                    
+            // Required to avoid memory leak for BTPayPalClient
+            self.authenticationSession = nil
+            if let error = error as? NSError {
+                switch error {
+                case ASWebAuthenticationSessionError.canceledLogin:
                     // User canceled by breaking out of the PayPal browser switch flow
                     // (e.g. System "Cancel" button on permission alert or browser during ASWebAuthenticationSession)
+                    if self.returnedToAppAfterPermissionAlert == true {
+                        // User tapped system cancel button on permission alert
+                        self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.webSessionAlertCanceled)
+                    }
+                    // general login cancel message for both user cancel
+                    self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserLoginCanceled)
                     completion(nil, BTPayPalError.canceled)
                     return
+                default:
+                    self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserLoginFailed)
+                    self.apiClient.sendAnalyticsEvent(BTPayPalAnalytics.tokenizeFailed)
+                    completion(nil, BTPayPalError.webSessionError(error))
+                    return
                 }
-
-                self.handleBrowserSwitchReturn(callbackURL, paymentType: paymentType, completion: completion)
             }
+            
+            self.handleBrowserSwitchReturn(callbackURL, paymentType: paymentType, completion: completion)
+        }
         
         authenticationSession?.presentationContextProvider = self
         returnedToAppAfterPermissionAlert = false
+        apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserPresentationStarted)
         isAuthenticationSessionStarted = authenticationSession?.start() ?? false
-
-        let authenticationSessionStatus = isAuthenticationSessionStarted ? "succeeded" : "failed"
-        apiClient.sendAnalyticsEvent("ios.\(paymentType.stringValue).authsession.start.\(authenticationSessionStatus)")
+        
+        if isAuthenticationSessionStarted {
+            apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserPresentationSucceeded)
+        } else {
+            apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserPresentationFailed)
+            apiClient.sendAnalyticsEvent(BTPayPalAnalytics.tokenizeFailed)
+            completion(nil, BTPayPalError.webSessionFailedToLaunch)
+            return
+        }
```

Comment:
take it or leave it, since we return here I think we could do:
```suggestion
        if !isAuthenticationSessionStarted {
            apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserPresentationFailed)
            apiClient.sendAnalyticsEvent(BTPayPalAnalytics.tokenizeFailed)
            completion(nil, BTPayPalError.webSessionFailedToLaunch)
            return
        }
        
        apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserPresentationSucceeded)
```